python/ray/data/tests/test_parquet.py

```diff
@@ -1327,6 +1327,28 @@ def test_write_with_schema(ray_start_regular_shared, tmp_path):
     assert pq.read_table(tmp_path).schema == schema
 
 
+@pytest.mark.parametrize(
+    "row_data",
+    [
+        [{"a": 1, "b": None}, {"a": 1, "b": 2}],
+        [{"a": None, "b": 3}, {"a": 1, "b": 2}],
+        [{"a": None, "b": 1}, {"a": 1, "b": None}],
+    ],
+    ids=["row1_b_null", "row1_a_null", "row_each_null"],
+)
+def test_write_auto_infer_nullable_fields(tmp_path, ray_start_regular_shared, row_data):
```

Comment:
Add the `restore_data_context` fixture so that changes aren't persisted across tests
```suggestion
def test_write_auto_infer_nullable_fields(tmp_path, ray_start_regular_shared, row_data, restore_data_context):
```